topanswers.js

```diff
@@ -1,36 +1,77 @@
-        return el.content.childNodes;
-    };
+    const htmlFromString = (html) => {
+        const el = document.createElement('template')
+        el.innerHTML = html
+        return el.content.childNodes
+    }
+    const updateList = (data, id) => {
+        data.forEach((site) => {
+            let template = [
+                `<div class="widget--body topanswers-site">
+            <img src="${site.logo_url}" height="32" width="32" alt />
+            <a href="${site.canonical_url}" target="_blank" rel="noopener noreferrer">${site.name}</a>
+        </div>`,
+                `<div class="grid--cell is-4 is-6-md is-12-sm">
+    <div class="widget">
+        <div class="widget--header is-complex">
+            <div class="has-text-align-center has-font-weight-bold has-font-size-display">
+                <a href="${site.canonical_url}"
+                    style="display: flex; align-items: center; justify-content: center;">
+                        <img src="${site.logo_url}" alt="Meta" style="height: 1.5em">
+                </a>
+            </div>
+        </div>
+        <div class="widget--body">
+            <p>${site.description}</p>
+        </div>
+        <div class="widget--footer">
+            <a href="${site.canonical_url}" class="button is-filled">Visit the site</a>
+        </div>
+    </div>
+</div>`,
+            ]
+            const elements = htmlFromString(template[id])
+            elements.forEach((el) =>
+                id
+                    ? codidact_list.appendChild(el)
+                    : topanswers_list.appendChild(el)
+            )
+        })
+    }
 
-    const API_URL = 'https://topanswers.xyz/communities.json';
-    const list = document.querySelector('.js-topanswers-list');
+    const topAns_URL = 'https://topanswers.xyz/communities.json'
+    const codidact_URL = './communities.json'
+    const codidact_list = document.querySelector('.community-list')
+    const topanswers_list = document.querySelector('.js-topanswers-list')
 
-    const resp = await fetch(API_URL, {
+    const resp2 = await fetch(codidact_URL, {
         headers: {
-            'Accept': 'application/json'
-        }
-    });
-
-    if (resp.status === 200) {
-        const data = await resp.json();
-        data.forEach(site => {
-            const template = `<div class="widget--body topanswers-site">
-                <img src="${site.logo_url}" height="32" width="32" alt />
-                <a href="${site.canonical_url}" target="_blank" rel="noopener noreferrer">${site.name}</a>
-            </div>`;
-            const elements = htmlFromString(template);
-            elements.forEach(el => list.appendChild(el));
-        });
+            Accept: 'application/json',
+        },
+    })
+    if (resp2.status === 200) {
+        const data = await resp2.json()
+        //add communities from json excluding first three communities (already existing for JS disabled users)
+        updateList(data.slice(3), 1)
```

Comment:
Why the `data.slice(3)`?

Reply:
the json file comes with 3 communities  so we skip those and other items   because  we already have those  inside html by default for  users with JS disabled

Reply:
In that case, the 3 should be counted dynamically from the HTML, rather than set as a static magic number. If we update the communities and forget to update this, that's a bug waiting to happen.

Reply:
we just add the new communities to the array in `communities.json` file

Reply:
Yes, but then when we add them to the HTML, we get a bug.

Reply:
we don't add to html we just add to community.json     ,
do you want to prepend new items to keep meta last

Reply:
We'll need to add them to the HTML too, otherwise we run into the same problem I was talking about yesterday for users without JS.